ksql-engine/src/main/java/io/confluent/ksql/structured/SchemaKGroupedStream.java

```diff
@@ -87,7 +101,8 @@
       final MaterializedFactory materializedFactory
   ) {
     this.kgroupedStream = kgroupedStream;
-    this.schema = schema;
+    this.sourceStep = sourceStep;
```

Comment:
Why no null check?

Reply:
Same reason as for kGroupedStream. It's possible for it to be null because of our usage of inheritance here. SchemaKGroupedTable extends SchemaKGroupedStream, but it doesn't have a source step that produces a grouped stream. We should at some point clean these up to not inherit and just implement a common interface.

Reply:
Something we should look to fix!